### ISSUE TITLE
DS Prefill :: Remove typecasts from DEVICE_FP32 MoE gate

### ISSUE THREAD

**Issue by fbajraktariTT** [tenstorrent/tt-metal#50239]
Both typecast from `_device_grouped_gate_fp32` method are very short ops (2-5us) which creates huge op-to-op gaps due to the fact that fast-dispatch cannot keep up with the obs. Thus, we should remove these two typecast by enabling `MoeGroupedTopkDeviceOperation` to upcast `logits` and `bias` directly in the kernel.

<img width="1228" height="679" alt="Image" src="https://github.com/user-attachments/assets/bac7533c-f2e8-4d12-b5b1-a06c93d4e49e" />
